src/driver/communication_interface.cc

```diff
@@ -416,47 +436,69 @@ void CommunicationInterface::HandlePlan(
-      piecewise_polynomial.value(piecewise_polynomial.start_time());
-
-  auto q = this->GetRobotState().q;
-  // TODO(@anyone): move this check to franka plan runner
-  Eigen::VectorXd q_eigen = utils::v_to_e(utils::ArrayToVector(q));
-
-  auto max_angular_distance =
-      utils::max_angular_distance(commanded_start, q_eigen);
-  if (max_angular_distance > params_.kMediumJointDistance) {
-    // discard the plan if we are too far away from current robot start
-    Eigen::VectorXd joint_delta = q_eigen - commanded_start;
-    dexai::log()->error(
+    dexai::log()->info(
         "CommInterface:HandlePlan: "
-        "discarding plan {}, mismatched start position with delta: {}.",
-        robot_spline->utime, joint_delta.transpose());
-    new_plan_buffer_.plan.reset();
-    lock.unlock();
-    PublishPlanComplete(robot_spline->utime, false /*  = failed*/,
-                        "mismatched_start_position");
-    return;
+        "plan {}, start time: {:.2f}, end time: {:.2f}",
+        robot_spline->utime, piecewise_polynomial.start_time(),
+        piecewise_polynomial.end_time());
+
+    // Start position == goal position check
+    // TODO(@anyone): change to append initial position and respline here
+    Eigen::VectorXd commanded_start =
+        piecewise_polynomial.value(piecewise_polynomial.start_time());
+
+    auto q = this->GetRobotState().q;
+    // TODO(@anyone): move this check to franka plan runner
+    Eigen::VectorXd q_eigen = utils::v_to_e(utils::ArrayToVector(q));
+
+    auto max_angular_distance =
+        utils::max_angular_distance(commanded_start, q_eigen);
+    if (max_angular_distance > params_.kMediumJointDistance) {
+      // discard the plan if we are too far away from current robot start
+      Eigen::VectorXd joint_delta = q_eigen - commanded_start;
+      dexai::log()->error(
+          "CommInterface:HandlePlan: "
+          "discarding plan {}, mismatched start position with delta: {}.",
+          robot_spline->utime, joint_delta.transpose());
+      new_plan_buffer_.plan.reset();
+      lock.unlock();
+      PublishPlanComplete(robot_spline->utime, false /*  = failed*/,
+                          "mismatched_start_position");
+      return;
+    }
+    new_plan_buffer_.plan = std::make_unique<PPType>(piecewise_polynomial);
+
+    if (new_plan_buffer_.exec_opt
+        == robot_msgs::plan_exec_opts_t::MOVE_UNTIL_STOP) {
+      new_plan_buffer_.contact_expected =
+          utils::ToRigidTransform(robot_spline->cartesian_goal).translation();
+    }
+
+  } else {
+    // Piecewise pose
+    auto X_EEcurrent_EEdesired {
+        utils::ToRigidTransform(robot_spline->cartesian_goal)};
+
+    std::vector<double> times_vec {0, 10.0};
+    auto cartesian_plan {PosePoly::MakeCubicLinearWithEndLinearVelocity(
+        times_vec,
+        {drake::math::RigidTransformd::Identity(), X_EEcurrent_EEdesired})};
+
+    new_plan_buffer_.cartesian_plan =
+        std::make_unique<PosePoly>(cartesian_plan);
```

Comment:
Oh nvm. We actually are making a piecewise cartesian plan?

Reply:
Yeah, but have I deleted that code from this PR